packages/terra-menu/src/Menu.jsx

```diff
@@ -113,11 +114,31 @@ class Menu extends React.Component {
   push(item) {
     this.setState((prevState) => {
       const newStack = prevState.stack.slice();
-      newStack.push(item);
+      let updatedStack = {};
```

Comment:
nit: this should be array or empty
```suggestion
      let updatedStack = [] OR let updatedStack;
```